src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/SeriesFields.scala

```diff
@@ -1,19 +1,34 @@
 package uk.gov.nationalarchives.tdr.api.graphql.fields
 
+import sangria.marshalling.circe._
+import io.circe.generic.auto._
 import uk.gov.nationalarchives.tdr.api.graphql.ConsignmentApiContext
 import uk.gov.nationalarchives.tdr.api.graphql.Arguments.BodyArg
 import sangria.macros.derive._
-import sangria.schema.{Field, ListType, ObjectType, fields}
+import sangria.schema.{Argument, Field, InputObjectType, ListType, ObjectType, fields}
 import uk.gov.nationalarchives.tdr.api.graphql.Tags.ValidateBody
 
 object SeriesFields {
   case class Series(seriesid: Long, bodyid: Option[Long] = None, name: Option[String] = None, code: Option[String] = None, description: Option[String] = None)
+  case class AddSeriesInput(bodyid: Option[Long] = None,
+                            name: Option[String] = None, code: Option[String] = None, description: Option[String] = None)
   implicit val SeriesType: ObjectType[Unit, Series] = deriveObjectType[Unit, Series]()
+  implicit val AddSeriesInputType: InputObjectType[AddSeriesInput] = deriveInputObjectType[AddSeriesInput]()
+
+  private val SeriesInputArg = Argument("addSeriesInput", AddSeriesInputType)
 
   val queryFields: List[Field[ConsignmentApiContext, Unit]] = fields[ConsignmentApiContext, Unit](
     Field("getSeries", ListType(SeriesType),
       arguments=BodyArg :: Nil,
       resolve = ctx => ctx.ctx.seriesService.getSeries(ctx.arg(BodyArg)),
       tags=List(ValidateBody()))
   )
+
+  val mutationFields: List[Field[ConsignmentApiContext, Unit]] = fields[ConsignmentApiContext, Unit](
+    Field(
+      "addSeries",
+      SeriesType,
+      arguments = List(SeriesInputArg),
+      resolve = ctx => ctx.ctx.seriesService.addSeries(ctx.arg(SeriesInputArg)))
```

Comment:
What's the plan for authenticating and authorizing requests to this endpoint? If we're doing this properly, we should set up the Lambda/Jenkins job that will call this endpoint, and authenticate it with Keycloak.

My suggestion for populating the dropdown was to just add a migration with some fake seed data, and add the real population endpoint later.

Reply:
Yes. There does need to be some authentication for the end point. I've asked Sam about what we should do for the authentication, ie a new role. OK. I thought we needed to add the endpoint to add the data. At least it will be ready for later.

Reply:
The way I've been writing it, there are two roles, `tdr_user` and `tdr_admin` The former being a normal user. 
I've added a commit https://github.com/nationalarchives/tdr-consignment-api/pull/4/commits/de562462d116e67056e166b9b319b19955f817f8 which adds a validator to check if the user is an admin user and rejects the request if they're not. There's a route unit test in there to show it working. 
We can have as many roles as we want within keycloak but for now I think user and admin are enough. If not, we can change the validation case class easily enough.

Reply:
We'll eventually need at least one machine client role that the backend tasks will use, but we might not need it yet.

So is the plan for now that one of us will get an admin token, and then call this mutation from Postman (or similar)? How do we get a token?

I've added a card to remind us to write a proper background task when we come to add real series: https://github.com/orgs/nationalarchives/projects/20#card-32933151

Reply:
Locally I was just printing the access token to console and using that in postman. That's probably not going to work so well in the proper environments thinking about it. Will write a database migration to add some seed data to the series and body table instead.

Reply:
Seed migrations added for series and body tables: https://github.com/nationalarchives/tdr-consignment-api-data/tree/master/lambda/src/main/resources/db/migration